Philips.CodeAnalysis.MsTestAnalyzers/TestShouldUseDisplayNameAnalyzer.cs

```diff
@@ -0,0 +1,156 @@
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+		protected override TestMethodImplementation OnInitializeTestMethodAnalyzer(AnalyzerOptions options, Compilation compilation, MsTestAttributeDefinitions definitions)
+		{
+			return new TestShouldUseDisplayName(definitions, Helper);
+		}
+
+		public class TestShouldUseDisplayName : TestMethodImplementation
+		{
+			public TestShouldUseDisplayName(MsTestAttributeDefinitions definitions, Helper helper) : base(definitions, helper)
+			{ }
+
+			protected override void OnTestMethod(SyntaxNodeAnalysisContext context, MethodDeclarationSyntax methodDeclaration, IMethodSymbol methodSymbol, bool isDataTestMethod)
+			{
+				if (isDataTestMethod)
+				{
+					CheckDataTestMethodForDisplayName(context, methodDeclaration);
+				}
+				else
+				{
+					CheckTestMethodForDescription(context, methodDeclaration);
+				}
+			}
+
+			private void CheckDataTestMethodForDisplayName(SyntaxNodeAnalysisContext context, MethodDeclarationSyntax methodDeclaration)
+			{
+				// Check for DataRow attributes with comments but no DisplayName
+				System.Collections.Generic.IEnumerable<AttributeSyntax> dataRowAttributes = methodDeclaration.AttributeLists
+					.SelectMany(list => list.Attributes)
+					.Where(attr => Helper.ForAttributes.IsDataRowAttribute(attr, context));
+
+				foreach (AttributeSyntax attribute in dataRowAttributes)
+				{
+					// Check if this DataRow has a comment but no DisplayName
+					var hasDisplayName = attribute.ArgumentList?.Arguments.Any(arg =>
+						arg.NameEquals?.Name.Identifier.ValueText == "DisplayName") == true;
+
+					if (!hasDisplayName)
+					{
+						// Look for trailing comment on the same line
+						var comment = GetTrailingComment(attribute);
+						if (!string.IsNullOrWhiteSpace(comment))
+						{
+							var diagnostic = Diagnostic.Create(Rule, attribute.GetLocation());
+							context.ReportDiagnostic(diagnostic);
+						}
+					}
+				}
+			}
+
+			private void CheckTestMethodForDescription(SyntaxNodeAnalysisContext context, MethodDeclarationSyntax methodDeclaration)
+			{
+				// Check if test method has Description attribute
+				var hasDescription = Helper.ForAttributes.HasAttribute(methodDeclaration.AttributeLists, context, MsTestFrameworkDefinitions.DescriptionAttribute, out _, out _);
+
+				if (!hasDescription)
+				{
+					// Look for leading comment before the method
+					var comment = GetLeadingComment(methodDeclaration);
+					if (!string.IsNullOrWhiteSpace(comment))
+					{
+						var diagnostic = Diagnostic.Create(Rule, methodDeclaration.Identifier.GetLocation());
+						context.ReportDiagnostic(diagnostic);
+					}
+				}
+			}
+
+			private string GetTrailingComment(AttributeSyntax attribute)
```

Comment:
Suggest to rename this method to include `Meaningful`. It implements a couple of rules that go beyond just "getting" a comment from the `AttributeSyntax`